src/Shouldly/MessageGenerators/ShouldBeInOrderMessageGenerator.cs

```diff
@@ -0,0 +1,18 @@
+namespace Shouldly.MessageGenerators
+{
+    internal class ShouldBeInOrderMessageGenerator : ShouldlyMessageGenerator
+    {
+        public override bool CanProcess(IShouldlyAssertionContext context)
+        {
+            return context.ShouldMethod == "ShouldBeInOrder";
+        }
+
+        public override string GenerateErrorMessage(IShouldlyAssertionContext context)
+        {
+            return
+$@"{context.CodePart}
+    should be in {context.Expected.ToString().ToLower()} order
+    but item at index {context.Key} was not.";
```

Comment:
A few possible suggestions:

1. You can introduce another property on the context for the SortOrder rather than hijacking Expected. This makes it easier to understand what is going on.
2. If you do that, then you can actually write the value of the out of place item
3. It would be nice to change the message to indicate that it is the first out of place item, `should be in ascending order but was not. The first out of order item found at index 1`

Another way to solve this problem is:

`unordered.OrderBy(i => i).ShouldBe(unordered)` - I think this extension can improve on the error message though.